### PR TITLE
Database analyzers (pg, mysql, redis)

### PR DESCRIPTION
**Pull request by marccampbell**
This adds a new postgres collector and analyzer that will determine if a connection URI is reachable and if the version is supported.

example:

```yaml
apiVersion: troubleshoot.replicated.com/v1beta1
kind: Preflight
metadata:
  name: sample
spec:
  collectors:
    - postgres:
        collectorName: pg
        uri: postgres://testuser:password@postgresql:5432/testdb?sslmode=disable
  analyzers:
    - postgres:
        checkName: Must be postgres 10.x or later
        collectorName: pg
        outcomes:
          - fail:
              when: "connected == false"
              message: Cannot connect to postgres server
          - fail:
              when: "version < 10.x"
              message: The postgres server must be at least version 10
          - pass:
              message: The postgres connection checks out
  ```

### CODE REVIEW

**Comment by marccampbell** [comment]
@GraysonNull will you add an icon the to spritesheet for this?

**Comment by GraysonNull** [comment]
Yeah I’ll get one made up first thing

**Review by divolgin — APPROVED** [review]

**Comment by dexhorthy** [comment]
cooooooool 👀

File: pkg/analyze/database_shared.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	"strconv"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/troubleshoot/pkg/collect"
+)
+
+func compareDatabaseConditionalToActual(conditional string, result *collect.DatabaseConnection) (bool, error) {
+	parts := strings.Split(strings.TrimSpace(conditional), " ")
```
Comment (divolgin) [inline pkg/analyze/database_shared.go:14]: I know this is not new, but `strings.Fields` is a better function to use for this.

**Review by divolgin — APPROVED** [review]

File: examples/preflight/mysql.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+apiVersion: troubleshoot.replicated.com/v1beta1
+kind: Preflight
+metadata:
+  name: sample
+spec:
+  collectors:
+    - mysql:
+        collectorName: my-mysql
+        uri: postgresql://doadmin:nokq949a9dom9qyx@postgres-test-do-user-1634526-0.a.db.ondigitalocean.com:25060/defaultdb?sslmode=require
```
Comment (laverya) [inline examples/preflight/mysql.yaml]: this looks a little more 'real' than the others

File: examples/preflight/redis.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  analyzers:
+    - redis:
+        checkName: Must be redis 5.x or later
+        collectorName: my-redis
+        outcomes:
+          - fail:
+              when: "connected == false"
+              message: Cannot connect to postgres server
+          - fail:
+              when: "version < 5.0.0"
```
Comment (laverya) [inline examples/preflight/redis.yaml:19]: I'm not a _huge_ fan of the magic-string system here
but it works, I suppose

File: pkg/collect/database_shared.go [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package collect
+
+type DatabaseConnection struct {
+	IsConnected bool   `json:"isConnected"`
+	Error       string `json:"error"`
```
Comment (laverya) [inline pkg/collect/database_shared.go]: should this omitempty?

**Review by laverya — APPROVED** [review]

File: examples/preflight/mysql.yaml (hunk 2 above)
Comment (marccampbell) [reply examples/preflight/mysql.yaml]: :+1: deleted and removed.  this was a e2e test (empty) server, shouldn't have been added but no worries if this connection string is out

**Comment by dexhorthy** [comment]
@marccampbell did this ever make it into reference docs? I can't seem to find it

**Pull request merged**